stock_by_warehouse_mrp_bom/views/mrp_bom_view.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="mrp_bom_line_view_form_inherit" model="ir.ui.view">
+        <field name="name">mrp.bom.line.stock.warehouse</field>
+        <field name="model">mrp.bom.line</field>
+        <field name="inherit_id" ref="mrp.mrp_bom_line_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//form/sheet/group" position="before">
+                <div class="text-center alert alert-info" role="alert">
+                    <p>The stock <span attrs="{'invisible':[('warehouse_id','=',False)]}">in: <field name="warehouse_id"/></span> that is immediately available is:
```

Comment:
How could be the warehouse empty?

It's a required field and it's filled by default.

Reply:
The field `warehouse_id` is related to the one in the [`picking_type_id` of the bom](https://github.com/odoo/odoo/blob/a33e04df6/addons/mrp/models/mrp_bom.py#L57) and that field is not required, even in the [view](https://github.com/odoo/odoo/blob/38e5d002ac/addons/mrp/views/mrp_bom_views.xml#L138) it says that can be invisible depending on the `type`, so if there is no other field in the BOM where the warehouse can be obtained it means that the warehouse can be empty

Reply:
Thanks a lot @CarmenMiranda, I completely forgot we were talking about BoMs,m, not sale orders. Resolving.

Reply:
Hahaha, yes is a BoM, personally, I haven't worked with this module, so I really don't know much about it hahaha, I just used the code to understand it, and I didn't found another way to get the warehouse 🙈